### PR TITLE
New parent endpoints

### PR DESCRIPTION
**Pull request by cr2david**
completed pages:
clan profile
create quest
parent profile

### CODE REVIEW

File: android/app/src/main/java/com/chorestory/adapter/ChildRecyclerViewAdapter.java [hunk 1]
```diff
@@ -10,15 +10,24 @@
...
+import com.chorestory.templates.ClanChildrenResponse.Children;
 
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 
 public class ChildRecyclerViewAdapter extends RecyclerView.Adapter<ChildRecyclerViewAdapter.MyViewHolder> {
 
-    private List<ChildRecyclerViewItem> itemList;
+    private ArrayList<ChildRecyclerViewItem> childrenList = new ArrayList<>();
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/adapter/ChildRecyclerViewAdapter.java]: private List

File: android/app/src/main/java/com/chorestory/app/CreateClanActivity.java [hunk 2]
```diff
@@ -31,6 +31,7 @@
     private String clanName;
     private String name;
     private String password;
+    private Integer picture;
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/app/CreateClanActivity.java]: private int

File: android/app/src/main/java/com/chorestory/fragment/ParentClanFragment.java [hunk 3]
```diff
@@ -68,4 +76,67 @@ public void onClick(View v) {
...
+                            ClanResponse.Data respData = response.body().getData();
+
+                            // Set the clan Title
+                            clanNameTextView.setText(getString(R.string.clan_title, respData.getClanName()));
+
+                            // Set the parent list
+                            List<ClanResponse.Data.Parent> parentList = respData.getParents();
+                            parentAdapter = new ParentRecyclerViewAdapter(parentList);
+                            parentRecyclerView.setAdapter(parentAdapter);
+
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/fragment/ParentClanFragment.java]: remove empty line

File: android/app/src/main/java/com/chorestory/fragment/ParentClanFragment.java [hunk 4]
```diff
@@ -68,4 +76,67 @@ public void onClick(View v) {
...
+                    public void onFailure(Call<ClanChildrenResponse> call, Throwable t) {
+                        Toaster.showToast(getContext(), "Internal error occurred.");
+                        // TODO: delete the token we have stored and redirect the user to the login page?
+                    }
+                });
+            } else {
+                // TODO: redirect to login page
+            }
+        }
+
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/fragment/ParentClanFragment.java]: remove empty line

File: android/app/src/main/java/com/chorestory/fragment/ParentCreateFragment.java [hunk 5]
```diff
@@ -19,13 +20,38 @@
...
+import retrofit2.Callback;
+import retrofit2.Response;
 
 public class ParentCreateFragment extends Fragment {
 
+    @Inject
+    RetrofitInterface retrofitInterface;
+    @Inject
+    TokenHandler tokenHandler;
+
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/fragment/ParentCreateFragment.java]: remove empty line

File: android/app/src/main/java/com/chorestory/fragment/ParentProfileFragment.java [hunk 6]
```diff
@@ -73,6 +86,39 @@ public void onClick(View v) {
...
+
+                            AccountResponse.Data respData = response.body().getData();
+
+                            // Set data for page
+                            parentNameTextView.setText(respData.getName());
+                            clanNameTextView.setText(respData.getClanName());
+                            parentEmailTextView.setText(respData.getEmail());
+
+                            parentImageView.setImageResource(R.drawable.king_color); // TODO: get image id
+
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/fragment/ParentProfileFragment.java]: remove empty line

File: android/app/src/main/java/com/chorestory/model/ChildRecyclerViewItem.java [hunk 7]
```diff
@@ -1,85 +1,43 @@
 package com.chorestory.model;
 
+
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/model/ChildRecyclerViewItem.java]: remove empty line

File: android/app/src/main/java/com/chorestory/templates/RegisterRequest.java [hunk 8]
```diff
@@ -5,11 +5,13 @@
     private String clan_name;
     private String name;
     private String password;
+    private Integer picture;
 
-    public RegisterRequest(String email, String clanName, String name, String password) {
+    public RegisterRequest(String email, String clanName, String name, String password, Integer picture) {
```
Comment (jimzhaozhengwang) [inline android/app/src/main/java/com/chorestory/templates/RegisterRequest.java]: int picture

File: android/app/src/main/java/com/chorestory/fragment/ParentCreateFragment.java [hunk 9]
```diff
@@ -271,4 +339,78 @@ private void enableButtons() {
...
+                                    Log.d("BUG", "Our child list is inconsistent somehow.");
+                                }
+                            }
+
+                            @Override
+                            public void onNothingSelected(AdapterView<?> parent) {
+                                selectedChildId = 0;
+                            }
+                        });
+
```
Comment (keller00) [inline android/app/src/main/java/com/chorestory/fragment/ParentCreateFragment.java]: Delete this empty line before @jimzhaozhengwang sees it 😄

**Pull request merged**